subprojects/core/src/main/java/org/gradle/api/internal/CompositeMutationGuard.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal;
+
+import org.gradle.api.Action;
+import org.gradle.api.specs.Spec;
+import org.gradle.internal.Factory;
+import org.gradle.util.CollectionUtils;
+
+import java.util.List;
+
+public class CompositeMutationGuard extends AbstractMutationGuard {
```

Comment:
When do we actually use this? It looks like all of the calls to `MutationGuards.of()` take a single target. I'm not sure why we would ever have more than one target for a given mutation.